examples_utils/benchmarks/command_utils.py

```diff
@@ -12,6 +12,19 @@
 logger = logging.getLogger(__name__)
 
 
+def determine_variant_timeout(global_timeout, benchmark_dict):
+    """Determine timeout for a benchmark variant,
+    using the global timeout and variant-specific timeout, if specified.
+    """
+    app_timeout = benchmark_dict.get("timeout")
+    if global_timeout is None:
+        return app_timeout
+    elif app_timeout is None:
+        return global_timeout
+    else:
+        return min(global_timeout, app_timeout)
```

Comment:
If think that min function here is not appropriate.
This way we are not able to override higher level timeout.
I think that we should return first the most local/close timeout.
This way: 
1. app_timeout if provided 
2. otherwise global_timeout if provided
3. otherwise default timeout.

Ah, it seems that there is no default timeout.
It should be introduced I think. It could be e.g. 2h.
If someone needs more then it should be overridden by app or global timeout.
This way we always save time and make the run time deterministic.

Reply:
That would make this a breaking change for other users (training apps) of the benchmarking tool. This tool is used for orchestrating very long running training (hours to days) so a default doesn't really make sense here.

Reply:
@michaln-gc The intention for this PR was to not break existing functionality. Since setting a default global timeout would require input and approval from owners of all the other apps that use this benchmarking script, it's a much bigger change than I was intending to make. Would you be ok if a separate task was created for that, and we land this PR in the meantime?

Reply:
ok

Reply:
Thanks, you're a blocking reviewer at the moment, could you approve if you're happy with the rest of the changes?